programs/aleo-verifier/src/processor.rs

```diff
@@ -18,50 +14,35 @@ use {
     std::str::FromStr,
 };
 
+const ALEO_VERIFIER: &str = "A1eoProof1111111111111111111111111111111111";
+
 pub struct Processor;
 
 impl Processor {
+    /// Accounts expected:
+    ///
+    /// 0. `[SIGNER]`: The account of the account initialise the verification
+    /// 1. `[WRITE]`: Aleo transactions verification results storage account
+    /// 2. `[]`: Aleo transaction data storage account
+    /// 3. `[]`: Aleo Program account
```

Comment:
I streamlined these accounts as well. I'm still not sure if the `Aleo Program account` would be really needed, since it's hardcoded any way, but after I got all the accounts, keys, seeds and everything aligned, I left this as-is for now.

Reply:
> we need the aleo_program account because we cannot create it in the onchain program

I don't think we need it? The address is hard coded into Solana node / on-chain program so it's known.

> ...and that the user needs to know that they are signing something that will invoke the native program set by aleo_program_id, which is used to generate the aleo_program account

But here you have a solid point. This would explain it. :+1: